cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -0,0 +1,233 @@
+                  console.log('Parsed ' + results.length + ' rows.');
+                  resolve(results);
+                });
+    });
+  }
+
+/// Parses rows and columns of SQL query into [SqlData].
+const parseSqlQuery = (data: Object): SqlData => {
+  let rows: any[] = [];
+  let cols: string[] = [];
+
+  if (data.columnMetadata != undefined) {
+    data.columnMetadata.map((value, _) => {
+      cols.push(value.name);
+    });
+  }
+
+  if (data.records != undefined) {
+    data.records.map((record: Array<Object>) => {
+      let row = {};
+      record.map((value, index) => {
+        if (value.stringValue !== 'undefined') {
+          row[cols[index]] = value.stringValue;
+        } else if (value.blobValue !== 'undefined') {
+          row[cols[index]] = value.blobValue;
+        } else if (value.doubleValue !== 'undefined') {
+          row[cols[index]] = value.doubleValue;
+        } else if (value.longValue !== 'undefined') {
+          row[cols[index]] = value.longValue;
+        } else if (value.booleanValue !== 'undefined') {
+          row[cols[index]] = value.booleanValue;
+        } else if (value.isNull) {
+          row[cols[index]] = null;
+        }
+      });
+      rows.push(row);
+    });
+    console.log('Found rows: ' + rows.length);
+  }
+  return {columns: cols, rows: rows};
+}
+
+/// Executes SQL statement argument against the MainCluster db.
+const executeSqlStatement = (secretArn: string, resourceArn: string,
+                             statement: string,
+                             db: String | undefined,
+                             callback: APIGatewayProxyCallback): void => {
+  let sqlParams = {
+    secretArn: secretArn,
+    resourceArn: resourceArn,
+    sql: statement,
+    database: db ?? 'postgres',  // Default db
+    includeResultMetadata: true
+  };
+
+  rdsDataService.executeStatement(
+    sqlParams, function (err: Error, data: Object) {
+      if (err) {
+        console.log(err);
+        callback('Error: RDS statement failed to execute');
+      } else {
+        console.log('Data is: ' + data);
+        callback(null, {statusCode: 200, body: JSON.stringify(data)});
+      }
+    });
+}
+/// Format [DemographicsTableRow] as SQL row.
+const formatPopulationTableRowAsSql = (row: DemographicsTableRow): string => {
+  let singleValue = [
+    row.censusGeoId, row.totalPopulation, row.percentageBlackPopulation(),
+    row.percentageWhitePopulation()
+  ];
+  return '(' + singleValue.join(',') + ')';
+}
+
+/// Write demographic data handler.
+exports.handler =
+  (event: APIGatewayEvent, context: Context,
+   callback: APIGatewayProxyCallback): void => {
+    const secretArn = event['secretArn'];
+    const mainClusterArn = event['mainClusterArn'];
+
+    const s3Params = {
+      Bucket: 'opendataplatformapistaticdata',
+      Key: 'alabama_acs_data.csv'
+    };
+    // Read CSV file and write to demographics table.
+    parseS3IntoDemographicsTableRow(s3Params)
```

Comment:
Can you use [`async/await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) syntax here instead of `.then`? It makes code more simple and readable.

Reply:
I did but ended up having to wrap it in a lot of nested promises which I think may have made it worse but at least it's all consistent now